notion-server/src/eth_magic.rs

```diff
@@ -0,0 +1,165 @@
+use std::default;
+use std::collections::{HashMap};
+
+#[derive(Debug)]
+pub struct EthMagicManager {
+    beacon_chain: BeaconChain,
+    shard_chains: Vec<ShardChain>,
+}
+
+impl EthMagicManager {
+    pub fn new() -> Self {
+        Self {
+            beacon_chain: BeaconChain::new(),
+            shard_chains: Vec::new(),
+        }
+    }
+
+    // Creates a new execution environment on the BeaconChain and returns the
+    // index of the created execution environment
+    pub fn add_new_execution_environment(&mut self, ee: CreateEeArgs) -> u32 {
```

Comment:
For consistency, I think the argument struct name should match the method name. In this case, perhaps `create_execution_environment`?

Reply:
This is fair feedback.  The only case where I think this may not apply 100% is if we have multiple functions that need `CreateEeArgs` as the input argument (eg. if we also have a function that builds the internal EE from the args, but doesn't save it to the chain).  In this case, though, I don't think we have that.  I'll update this to `create_execution_environment`.